### PR TITLE
feat: Automatic Summary feature DOCS-695

### PR DESCRIPTION
**Pull request by mrfyda**
### :eyes: Live preview
https://storyteller--docs-codacy.netlify.app/repositories-configure/integrations/github-integration/#generating-automatic-pull-request-summaries

### :construction: To do
-   [x] If relevant, include the Jira issue key at the end of the pull request title
-   [ ] Perform a self-review of the changes
-   [ ] Fix any issues reported by the CI/CD
-   [x] Depends on codacy/storyteller#18

### CODE REVIEW

File: docs/repositories-configure/integrations/github-integration.md [hunk 1]
```diff
@@ -79,6 +79,32 @@ Adds AI-enhanced comments, providing insights and ready-to-commit AI-generated f
...
+```yaml
+reviews:
+  high_level_summary: true
+```
+
+You can also enable this feature across your organization by creating the above file in the root of a repository named `.codacy`. (This file will be used as the default configuration for all repositories in the organization, and overriden by the repository-specific configuration file.)
+
+Once enabled, summaries will be created when pull requests are opened and kept up-to-date to reflect any changes to the pull request.
+
+Pull requests opened by Bot account (such as Dependabot) are automatically ignored.
```
Comment (afsmeira) [inline docs/repositories-configure/integrations/github-integration.md:106]:
```suggestion
Pull requests opened by Bot account (such as Dependabot) are automatically ignored.
```
```suggestion
Pull requests opened by Bot accounts (such as Dependabot) are automatically ignored.
```

File: docs/repositories-configure/integrations/github-integration.md [hunk 2]
```diff
@@ -79,6 +79,32 @@ Adds AI-enhanced comments, providing insights and ready-to-commit AI-generated f
 
 ![AI-enhanced comment on GitHub](images/github-integration-ai-comment.png)
 
+### Automatic summaries
+
+Even when not written by developers, Codacy can provide an high-level summary for each pull request.
```
Comment (afsmeira) [inline docs/repositories-configure/integrations/github-integration.md:84]: I'm not a fan of this sentence but I don't have a good suggestion.

Comment (mrfyda) [reply docs/repositories-configure/integrations/github-integration.md:84]:
```suggestion
Codacy can give a clear, high-level summary for each pull request, even if it's not written by developers.```

Comment (afsmeira) [inline docs/repositories-configure/integrations/github-integration.md:84]:
```suggestion
Codacy can provide a clear, high-level summary for each pull request, even if one is not written by developers.
```

Comment (afsmeira) [reply docs/repositories-configure/integrations/github-integration.md:84]: Yeah that's better, I just have a tiny suggestion
```suggestion
Codacy can provide a clear, high-level summary for each pull request, even if one is not written by developers.
```

**Review by afsmeira — APPROVED** [review]

File: docs/repositories-configure/integrations/github-integration.md [hunk 3]
```diff
@@ -79,6 +79,31 @@ Adds AI-enhanced comments, providing insights and ready-to-commit AI-generated f
...
+reviews:
+  high_level_summary: true
+```
+
+You can also enable this feature across your organization by creating the above file in the root of a repository named `.codacy`. This file will be used as the default configuration for all repositories in the organization and overridden by repository-specific configuration files.
+
+Once enabled, summaries will be created when pull requests are opened and updated at each commit to reflect any changes to the pull request.
+
+Pull requests opened by bots, such as Dependabot, are ignored.
+
```
Comment (mrfyda) [inline docs/repositories-configure/integrations/github-integration.md:106]:
```suggestion
### common problems(?)
seeing duplicated comments with the summary => likely cause is multiple configured webhooks to Codacy. solution: delete all but one webhook
```

File: docs/repositories-configure/integrations/github-integration.md [hunk 4]
```diff
@@ -79,6 +79,34 @@ Adds AI-enhanced comments, providing insights and ready-to-commit AI-generated f
...
+    This is an upcoming Codacy feature. If you're interested, ask us for early access.
+
+Codacy can provide a clear, high-level summary of the code changes introduced by a pull request, based on the committed code.
+Codacy generates an overview of the changes in the pull request so that any reviewer can understand its intent and impact.
+
+![Automatic Summary on GitHub](images/github-integration-automatic-summary.png)
+
+!!! note
+    -   This feature is only available for GitHub.
+    -   This feature uses [Amazon Bedrock](https://aws.amazon.com/bedrock/) within Codacy's existing infrastructure. No information is shared with any other third party or used to train AI models.
```
Comment (mrfyda) [inline docs/repositories-configure/integrations/github-integration.md:94]:
```suggestion
    -   This feature uses only AWS services within Codacy's existing infrastructure. No information is shared with any other third party or used to train AI models.
```

File: docs/repositories-configure/integrations/github-integration.md [hunk 5]
```diff
@@ -79,6 +79,34 @@ Adds AI-enhanced comments, providing insights and ready-to-commit AI-generated f
...
+  high_level_summary: true
+```
+
+You can also enable this feature across your organization by creating the above file in the root of a repository named `.codacy`. This file will be used as the default configuration for all repositories in the organization and overridden by repository-specific configuration files.
+
+Once enabled, summaries will be created when pull requests are opened and updated at each commit to reflect any changes to the pull request.
+
+Pull requests opened by bots, such as Dependabot, are ignored.
+If you see duplicated comments posted by Codacy on the same pull request, please ensure that your repository only has one configured webhook for Codacy.
 ## See also
```
Comment (mrfyda) [inline docs/repositories-configure/integrations/github-integration.md:110]:
```suggestion

## See also
```

File: docs/repositories-configure/integrations/github-integration.md [hunk 6]
```diff
@@ -79,6 +79,35 @@ Adds AI-enhanced comments, providing insights and ready-to-commit AI-generated f
...
+reviews:
+  high_level_summary: true
+```
+
+You can also enable this feature across your organization by creating the above file in the root of a repository named `.codacy`. This file will be used as the default configuration for all repositories in the organization and overridden by repository-specific configuration files.
+
+Once enabled, summaries will be created when pull requests are opened and updated at each commit to reflect any changes to the pull request.
+
+Pull requests opened by bots, such as Dependabot, are ignored.
+If you see duplicated comments posted by Codacy on the same pull request, please ensure that your repository only has one configured webhook for Codacy.
```
Comment (mrfyda) [inline docs/repositories-configure/integrations/github-integration.md:109]:
```suggestion

If you see duplicated comments posted by Codacy on the same pull request, please ensure that your repository only has one configured webhook for Codacy.
```

File: docs/repositories-configure/integrations/github-integration.md [hunk 7]
```diff
@@ -79,6 +79,36 @@ Adds AI-enhanced comments, providing insights and ready-to-commit AI-generated f
 
 ![AI-enhanced comment on GitHub](images/github-integration-ai-comment.png)
 
+## Generating automatic pull request summaries
+
+!!! info "This is a preview feature"
+    This is an upcoming Codacy feature. If you're interested, ask us for early access.
```
Comment (nicklem) [inline docs/repositories-configure/integrations/github-integration.md:85]:
```suggestion
    This is an upcoming Codacy feature. If you're interested, contact <a href="mailto:support@codacy.com">support@codacy.com</a> for early access.
```

**Review by nicklem — APPROVED** [review]
Left a final tweak. LGTM 👍

**Pull request merged**